ingestion/src/metadata/profiler/processor/core.py

```diff
@@ -603,7 +603,9 @@ def get_profile(self) -> CreateTableProfileRequest:
 
             if self._system_results:
                 system_profile = [
-                    SystemProfile(**system_result)
+                    system_result
+                    if isinstance(system_result, SystemProfile)
+                    else SystemProfile.model_validate(system_result)
```

Comment:
checked pydantic API. this works:

```python
from metadata.ingestion.models.custom_pydantic import BaseModel
class Model(BaseModel):
    a: int
    
m1 = Model.model_validate(Model(a=1))
m2 = Model.model_validate({"a": 1})
assert m1 == m2
```

so you can just use:

```suggestion
                    SystemProfile.model_validate(system_result)
```